File: api/src/main/java/kimosabe/api/audit/AuditorAwareImpl.java
package kimosabe.api.audit;

import kimosabe.api.model.User;
import org.springframework.data.domain.AuditorAware;
import org.springframework.security.authentication.AnonymousAuthenticationToken;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;

import java.util.Optional;

public class AuditorAwareImpl implements AuditorAware<String> {
    @Override
    public Optional<String> getCurrentAuditor() {
        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
        if (!auth.isAuthenticated() || auth instanceof AnonymousAuthenticationToken) {
            return Optional.empty();
        }
        return Optional.of(((User)auth.getPrincipal()).getUsername());
    }
}

File: api/src/main/kotlin/com/kimosabe/api/audit/AuditorAwareImpl.kt
package com.kimosabe.api.audit

import com.kimosabe.api.model.User
import org.springframework.data.domain.AuditorAware
import org.springframework.security.authentication.AnonymousAuthenticationToken
import org.springframework.security.core.context.SecurityContextHolder
import java.util.*

class AuditorAwareImpl : AuditorAware<UUID> {
  override fun getCurrentAuditor(): Optional<UUID> {
    val auth = SecurityContextHolder.getContext().authentication
    return if (!auth.isAuthenticated || auth is AnonymousAuthenticationToken) {
      Optional.empty()
    } else Optional.of((auth.principal as User).userId)
  }
}
